packages/shared/src/lib/share.ts

```diff
@@ -30,3 +30,47 @@ export const getTelegramShareLink = (link: string, text: string): string =>
   `https://t.me/share/url?url=${encodeURIComponent(link)}&text=${text}`;
 export const getEmailShareLink = (link: string, subject: string): string =>
   `mailto:?subject=${subject}&body=${encodeURIComponent(link)}`;
+
+export const getShareLink = (
+  provider: ShareProvider,
+  link: string,
+  text?: string,
```

Comment:
We usually use object arguments, easier to add new ones. Same below for `addLinkShareTrackingParams`